core/win32/callback.c

```diff
@@ -5438,6 +5438,15 @@ intercept_exception(app_state_at_intercept_t *state)
             cxt->CXT_XIP = (ptr_uint_t) dcontext->asynch_target;
             /* now handle the fault just like RaiseException */
             DODEBUG({ known_source = true; });
+        } else if ((app_pc) pExcptRec->ExceptionAddress ==
```

Comment:
Syscall can cause exception (the test case does this).
So we need to translate the address from the do_syscall method to the program address where the sys call happens

Reply:
I don't follow: the syscall itself should not cause an exception: it would just return a failure return value, not throw an exception.  Could you elaborate on why an exception would happen whose PC is the entry to our generated int 0x2e?

Reply:
Well, I do not know what it should make, but the syscall does throw an exception, see the test case.
I guess that this happens because the syscall parameters are invalid (0xdeadc0de)

Reply:
This doesn't make sense to me: this has never happened before.  Invalid parameters result in the kernel returning an appropriate error code, not throwing an exception into user mode.  Is this limited to WOW64 and this happens during the syscall argument marshalling in (64-bit) user mode and that's why there's an exception, because it's not yet in the kernel?

Reply:
But you're saying this exception is on an int 0x2e, not a WOW64 "syscall" gateway of the call* to the WOW64 library layer.